StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/ElementOrderCodeFixProvider.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                return document;
+            }
+
+            syntaxRoot = UpdateSyntaxRoot(memberDeclaration, syntaxRoot);
+
+            return document.WithSyntaxRoot(syntaxRoot);
+        }
+
+        private static SyntaxNode UpdateSyntaxRoot(MemberDeclarationSyntax memberDeclaration, SyntaxNode syntaxRoot)
+        {
+            var parentDeclaration = memberDeclaration.Parent;
+            var memberToMove = new MemberOrderHelper(memberDeclaration, checkElementType, checkAccessLevel, checkConst, checkStatic, checkReadonly);
+
+            if (parentDeclaration is TypeDeclarationSyntax)
+            {
+                return HandleTypeDeclaration(memberToMove, (TypeDeclarationSyntax)parentDeclaration, syntaxRoot);
+            }
+
+            if (parentDeclaration is NamespaceDeclarationSyntax)
+            {
+                return HandleNamespaceDeclaration(memberToMove, (NamespaceDeclarationSyntax)parentDeclaration, syntaxRoot);
+            }
+
+            if (parentDeclaration is CompilationUnitSyntax)
+            {
+                return HandleCompilationUnitDeclaration(memberToMove, (CompilationUnitSyntax)parentDeclaration, syntaxRoot);
+            }
+
+            return syntaxRoot;
+        }
+
+        private static SyntaxNode HandleTypeDeclaration(MemberOrderHelper memberOrder, TypeDeclarationSyntax typeDeclarationNode, SyntaxNode syntaxRoot)
+        {
+            return MoveMember(memberOrder, typeDeclarationNode.Members, syntaxRoot);
+        }
+
+        private static SyntaxNode HandleCompilationUnitDeclaration(MemberOrderHelper memberOrder, CompilationUnitSyntax compilationUnitDeclaration, SyntaxNode syntaxRoot)
+        {
+            return MoveMember(memberOrder, compilationUnitDeclaration.Members, syntaxRoot);
+        }
+
+        private static SyntaxNode HandleNamespaceDeclaration(MemberOrderHelper memberOrder, NamespaceDeclarationSyntax namespaceDeclaration, SyntaxNode syntaxRoot)
+        {
+            return MoveMember(memberOrder, namespaceDeclaration.Members, syntaxRoot);
+        }
+
+        private static SyntaxNode MoveMember(MemberOrderHelper memberOrder, SyntaxList<MemberDeclarationSyntax> members, SyntaxNode syntaxRoot)
+        {
+            foreach (var member in members)
+            {
+                var orderHelper = new MemberOrderHelper(member, checkElementType, checkAccessLevel, checkConst, checkStatic, checkReadonly);
+
+                if (orderHelper.Priority < memberOrder.Priority)
+                {
+                    syntaxRoot = MoveMember(syntaxRoot, memberOrder.Member, member);
+                    break;
+                }
+            }
+
+            return syntaxRoot;
+        }
+
+        private static SyntaxNode MoveMember(SyntaxNode root, MemberDeclarationSyntax field, MemberDeclarationSyntax firstNonConst)
+        {
+            var trackedRoot = root.TrackNodes(field, firstNonConst);
+            var fieldToMove = trackedRoot.GetCurrentNode(field);
```

Comment:
Update this variable name, you're moving members here no longer fields, same goes for the variables below